code/obj/item/mob_parts/robot_parts.dm

```diff
@@ -1216,6 +1212,257 @@ ABSTRACT_TYPE(/obj/item/parts/robot_parts/leg/right)
 /obj/item/parts/robot_parts/robot_frame/syndicate
 	syndicate = TRUE
 
+
+
+/obj/item/parts/robot_parts/drone_frame
+	name = "drone frame"
+	icon_state = "frame"
+	max_health = 5000
+	/// This will make the borg a syndie one
+	var/emagged = 0
+	var/freemodule = TRUE
+	var/build_step = 0
+	var/shelltypetoapply = "eyebot"
+	var/moduletoapply = null
+	var/obj/item/cell/cell = null
+	var/obj/item/ai_interface/ai_interface = null
+	var/obj/item/ = null
+	appearance_flags = KEEP_TOGETHER
+
+	var/image/image_plate_overlay = null
+	var/image/image_wire_overlay = null
+	var/image/image_radio_overlay = null
+	var/image/image_brain_overlay = null
+	var/image/image_interface_overlay = null
+	var/image/image_cell_overlay = null
+
+	New()
+		..()
+		src.icon_state = "frame-" + src.shelltypetoapply;
+		src.UpdateIcon()
+
+	attackby(obj/item/W, mob/user)
+		if (istype(W, /obj/item/sheet))
+			if (src.build_step < 1)
+				var/obj/item/sheet/M = W
+				if (M.change_stack_amount(-1))
+					src.build_step++
+					boutput(user, "You add the plating to [src]!")
+					playsound(src, 'sound/impact_sounds/Generic_Stab_1.ogg', 40, 1)
+					src.UpdateIcon()
+					return
+				else
+					boutput(user, "You need at least one metal sheet to add plating! How are you even seeing this message?! How do you have a metal sheet that has no metal sheets in it?!?!")
+					user.drop_item()
+					qdel(W) // no bizarro nega-sheets for you :v
+					return
+			else
+				boutput(user, "\The [src] already has plating!")
+				return
+
+		else if (istype(W, /obj/item/cable_coil))
+			if (src.build_step == 1)
+				var/obj/item/cable_coil/coil = W
+				if (coil.amount >= 3)
+					src.build_step++
+					boutput(user, "You add \the cable to [src]!")
+					playsound(src, 'sound/impact_sounds/Generic_Stab_1.ogg', 40, 1)
+					src.UpdateIcon()
+					coil.use(3)
+					if (coil.amount < 1)
+						user.drop_item()
+						qdel(coil)
+					return
+				else
+					boutput(user, "You need at least three lengths of cable to install it in [src].")
+					return
+			else if (src.build_step > 1)
+				boutput(user, "\The [src] already has wiring!")
+
+			else
+				boutput(user, "You need to plate the frame before you can add wires to it!")
+				return
```

Comment:
Prefer putting "fail" cases first, then just doing the behavior once they've been checked and early-returned from.